### ISSUE TITLE
Questions about training details

### ISSUE THREAD

**Issue by TenChickens666** [ipolharvard/ethos-ares#17]
Hi, thanks for sharing this great work! I have two questions about training ETHOS:

1. How do you handle MIMIC and MIMIC-ED datasets simultaneously during training?
I went through the code but didn't find where both datasets are used together. Currently, I only use the MIMIC dataset to train the model. Should they be combined in a specific way, or are they trained separately?

2. Training loss and validation loss values
When training on MIMIC, I observed that after around step 147,000 my training seems to stop decreasing (train loss ≈ 0.8176, val loss ≈ 0.8539).
Could you share roughly what training and validation loss you achieved at convergence?

Thank you again for your wonderful work!

**Comment by gabshi59** [comment]
Hi @prenc, thanks again for your great work and your reply!

I’ve trained the model for 192,000 steps, and achieved a validation loss of 0.84937, which seems reasonably low. However, when I evaluated the model by generating 7 repetitions on a 1% subset of the dataset, I only got an AUC of 0.7384 for ICU_MORTALITY, which is quite a bit lower than expected.

<img width="502" height="391" alt="Image" src="https://github.com/user-attachments/assets/6a93acff-44db-4a52-8caf-e9e1ff27c425" />

I'm wondering if you have any insights on this discrepancy. A few things I’m trying to figure out:

1. Is the validation loss directly correlated with downstream classification performance (e.g., AUC)? 

2. Should I continue the training for more steps? Is it normal to get such results at 192,000 steps?

3. Are there any tips to calculate the final AUC based on the repetition? Currently I just use majority voting.

Any suggestions or clarification would be greatly appreciated!

**Comment by prenc** [comment]
Can you first let me know what is your performance on downstream tasks and how many repetitions you run in each case? We don't report ICU mortality task in our ETHOS-ARES paper.

**Comment by prenc** [comment]
Hey @TenChickens666, thank you for the clarification. I haven’t run this task in a while, but I’ll check its performance with the current implementation of ETHOS. To test my models, I usually use 30day readmission because it’s the quickest task.

I’ll get back to you once I have more info.

**Comment by prenc** [comment]
Hi @TenChickens666, thanks for your patience.

I recently ran an extensive evaluation of several models across multiple downstream tasks to establish a checkpoint of the current performance. You can find the results here: [link](https://docs.google.com/spreadsheets/d/198wdPCYDPYflkKfc3wuS68lO6Ilec4IrZI3F_85RdXw/edit?usp=sharing). Please let me know if you have any questions.

Regarding your concern about ICU mortality: I have not been able to replicate the performance reported in the original paper. It's unclear to me whether this was a bug or a tokenization process that was particularly favorable for that task. However, if you use the code from the original deprecated repo, you should see results that are more or less consistent with the original findings.

On the positive side, we are now achieving stronger performance on readmission, while hospital mortality remains stable.

It's also worth noting that this is not a one-to-one comparison. I ran the evaluation on half of the original test set, and our methodology has since evolved, for example, we now run at least 100 repetitions. And the results are not the performance plateau. Additionally, these results come from the private ETHOS repo, which introduces significant differences in task evaluation compared to the older codebase.

**Comment by prenc** [comment]
Closing due to inactivity. Please reopen if case of any more questions in that thread.
